elf/Makefile

```diff
@@ -2,7 +2,7 @@ all: prog.uelf.o
 
 include ../Makefile.common
 
-CFLAGS   = -std=gnu11 -O0 -Wall -Werror -fno-builtin -ffreestanding
+CFLAGS   = -std=gnu11 -Og -Wall -Werror -fno-builtin -ffreestanding
```

Comment:
You can use command `add-symbol-file` [[1]](https://sourceware.org/gdb/current/onlinedocs/gdb/Files.html#Files) to tell gdb that that there is a file is loaded at the specified address. With some scripting it can even be made to work for dynamically loaded binaries (including DSOs in user space).